### PR TITLE
Some micro optimizations

### PR DESCRIPTION
**Pull request by Neui**
A while ago in September I tried to optimize the startup with many tasks currently unsuccessfully. In the meanwhile I did those micro optimizations, they shouldn't change much. See the commits for more info.

I haven't really tested them since them, but they are relatively simple enough.

BTW, in `dates.py`, at the file level, `FUNCS` is initialized with some `datetime` objects with being relative to "today", but AFAIK the value isn't updated after a new day. Not sure if this is significant (or where this is used), but I'll just point this out here.
```python
FUNCS = {
    NOW: datetime.date.today(),
    SOON: datetime.date.today() + datetime.timedelta(15),
    SOMEDAY: datetime.date.max,
    NODATE: datetime.date.max - datetime.timedelta(1),
}
```

### CODE REVIEW

File: GTG/gtk/browser/treeview_factory.py [hunk 1]
```diff
@@ -152,16 +152,33 @@ def get_task_closeddate_column_string(self, node):
             return ""
 
     def sort_by_startdate(self, task1, task2, order):
+        t1 = task1.get_start_date()
+        t2 = task2.get_start_date()
+        return self.__date_comp_continue(task1, task2, order, t1, t2)
         sort = self.__date_comp(task1, task2, 'start', order)
         return sort
```
Comment (diegogangl) [inline GTG/gtk/browser/treeview_factory.py:159]: Left an inaccessible return here

**Review by diegogangl — CHANGES_REQUESTED** [review]
Nice, seems to have made some things more simple too

File: GTG/gtk/browser/treeview_factory.py (hunk 1 above)
Comment (Neui) [reply GTG/gtk/browser/treeview_factory.py]: Fixed

**Review by diegogangl — APPROVED** [review]

**Comment by diegogangl** [comment]
Landed!

**Pull request merged**